vendor/github.com/prometheus/prometheus/model/labels/labels_stringlabels.go

```diff
@@ -412,7 +413,7 @@ func yoloBytes(s string) (b []byte) {
 // New returns a sorted Labels from the given labels.
 // The caller has to guarantee that all label names are unique.
 func New(ls ...Label) Labels {
-	slices.SortFunc(ls, func(a, b Label) bool { return a.Name < b.Name })
+	slices.SortFunc(ls, func(a, b Label) int { return strings.Compare(a.Name, b.Name) })
```

Comment:
I would recommend to consider replacing `strings.Compare()` in this file too, and run a benchmark to see the actual perf impact.